qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -212,6 +219,43 @@ def download_resource(self):
                         QUrl.fromLocalFile(str(Path(file_path).parent))
                     )
 
+    def add_model_to_qgis(self):
+        resource = self.selected_resource()
+        qgis_user_dir = QgsApplication.qgisSettingsDirPath()
+        default_model_path = Path(qgis_user_dir, "processing", "models")
+        if not default_model_path.exists():
+            default_model_path.mkdir(parents=True, exist_ok=True)
+
+        default_model_path = os.path.join(
+            default_model_path, os.path.basename(resource.file)
+        )
+
+        file_path = QFileDialog.getSaveFileName(
+            self,
+            self.tr("Save Model"),
+            str(default_model_path),
+            self.tr("QGIS Model (*.model3)"),
+        )[0]
+
+        if file_path:
+            if not download_resource_file(resource.file, file_path):
+                text = self.tr(f"Download failed for {resource.name}")
+                self.iface.messageBar().pushMessage(
+                    self.tr("Warning"), text, level=Qgis.Warning, duration=5
+                )
+            else:
+                text = self.tr(
```

Comment:
As you probably already know, we create a success message in several places (also a warning or error message). It's a good idea to put create a function and use it. DRY principle. But we can do it later or in another PR.